### PR TITLE
Fix unused warnings release

### PR DESCRIPTION
**Pull request by spham-amzn**
Fixes for 'unused variable' warnings -> errors in release builds. 

- Trace, Warnings, and Error macros are ifdef'd out in release, causing unused variables to only show up in those situations
- [[maybe_unused]] mostly used, but for the case of unused lambda captures, AZ_UNUSED will be used in its place

### CODE REVIEW

**Review by amznestebanpapp — COMMENTED** [review]
I thought you were going to add an `AZ_UNUSED` inside the `AZ_Warning`. `AZ_Error` and `AZ_Assert` macros, marking the condition unused.

File: Code/Framework/AzCore/AzCore/IO/Streamer/FullFileDecompressor.cpp [hunk 1]
```diff
@@ -45,8 +45,8 @@ namespace AZ
             }
         }
 
-        static constexpr char DecompBoundName[] = "Decompression bound";
-        static constexpr char ReadBoundName[] = "Read bound";
+        [[maybe_unused]] static constexpr char DecompBoundName[] = "Decompression bound";
+        [[maybe_unused]] static constexpr char ReadBoundName[] = "Read bound";
```
Comment (amznestebanpapp) [inline Code/Framework/AzCore/AzCore/IO/Streamer/FullFileDecompressor.cpp:49]: these should be wrapped in `#if AZ_STREAMER_ADD_EXTRA_PROFILING_INFO` instead

File: Code/Framework/AzCore/AzCore/IO/Streamer/Scheduler.cpp [hunk 2]
```diff
@@ -14,8 +14,8 @@
 
 namespace AZ::IO
 {
-    static constexpr char SchedulerName[] = "Scheduler";
-    static constexpr char ImmediateReadsName[] = "Immediate reads";
+    [[maybe_unused]] static constexpr char SchedulerName[] = "Scheduler";
+    [[maybe_unused]] static constexpr char ImmediateReadsName[] = "Immediate reads";
```
Comment (amznestebanpapp) [inline Code/Framework/AzCore/AzCore/IO/Streamer/Scheduler.cpp:18]: these should be wrapped in `#if AZ_STREAMER_ADD_EXTRA_PROFILING_INFO` instead

File: Code/Framework/AzCore/AzCore/IO/Streamer/StreamerContext.cpp [hunk 3]
```diff
@@ -17,9 +17,9 @@ namespace AZ
     namespace IO
     {
         static constexpr char ContextName[] = "Context";
-        static constexpr char PredictionAccuracyName[] = "Prediction accuracy (ms)";
-        static constexpr char LatePredictionName[] = "Early completions";
-        static constexpr char MissedDeadlinesName[] = "Missed deadlines";
+        [[maybe_unused]] static constexpr char PredictionAccuracyName[] = "Prediction accuracy (ms)";
+        [[maybe_unused]] static constexpr char LatePredictionName[] = "Early completions";
+        [[maybe_unused]] static constexpr char MissedDeadlinesName[] = "Missed deadlines";
```
Comment (amznestebanpapp) [inline Code/Framework/AzCore/AzCore/IO/Streamer/StreamerContext.cpp:22]: these should be wrapped in `#if AZ_STREAMER_ADD_EXTRA_PROFILING_INFO` instead

File: Code/Legacy/CrySystem/LocalizedStringManager.cpp [hunk 4]
```diff
@@ -32,8 +32,8 @@
 #define MAX_CELL_COUNT 32
 
 // CVAR names
-const char c_sys_localization_debug[] = "sys_localization_debug";
-const char c_sys_localization_encode[] = "sys_localization_encode";
+[[maybe_unused]] const char c_sys_localization_debug[] = "sys_localization_debug";
+[[maybe_unused]] const char c_sys_localization_encode[] = "sys_localization_encode";
```
Comment (amznestebanpapp) [inline Code/Legacy/CrySystem/LocalizedStringManager.cpp:36]: wrapped with `#if !defined(_RELEASE)` instead

**Review by amznestebanpapp — CHANGES_REQUESTED** [review]

**Comment by amznestebanpapp** [comment]
> I thought you were going to add an `AZ_UNUSED` inside the `AZ_Warning`. `AZ_Error` and `AZ_Assert` macros, marking the condition unused.

just realized that this would only cover the cases where the variable is in the condition, we would have to also mark unused variables that are passed as parameters

**Comment by spham-amzn** [comment]
> 
> 
> I thought you were going to add an `AZ_UNUSED` inside the `AZ_Warning`. `AZ_Error` and `AZ_Assert` macros, marking the condition unused.

No, that wasnt the plan. The macro used VARIADIC arguments, so i'm not sure how to do that

**Review by amznestebanpapp — APPROVED** [review]

**Review by AMZN-ScottR — APPROVED** [review]

**Pull request merged**